### PR TITLE
Adding 'Defining custom networks' doc

### PR DESCRIPTION
**Pull request by jessiemongeon1**
Thank you for your contribution to the IC Developer Portal.
Before submitting your Pull Request, please make sure that:

- [ ] You have updated the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file according to your changes.
- [ ] You have registered new documents to [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js) if any.

### CODE REVIEW

**Review by jennifertrin — APPROVED** [review]
Looks good to me! You may want to have another team member from engineering take a quick look

File: docs/references/networks-json.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    "bind": "localhost:4943",
+    "replica": {
+      "subnet_type": "application"
+    }
+  },
+    "network_2": {
+    "bind": "127.0.0.1:4943",
+    "replica": {
+      "subnet_type": "application"
+    }
```
Comment (dsarlis) [inline docs/references/networks-json.mdx:68]: Identation is a bit off for `network_2`.

File: docs/references/networks-json.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+To define a project-specific network, add a "networks" section to your `dfx.json` file such as:
+
+```json
+"networks": {
+    "dev": {
+      "providers": [
+        "https://ic0.app"
+      ],
+      "type": "persistent"
```
Comment (dsarlis) [inline docs/references/networks-json.mdx:108]: Maybe explain the options a little bit? E.g. in the above section, you explain that bind can be localhost or any other domain. The same (I believe) applies for "providers". Then we might want to say something about the "type" (persistent/ephemeral iirc).

**Review by dsarlis — APPROVED** [review]

File: docs/references/networks-json.mdx (hunk 2 above)
Comment (jessiemongeon1) [reply docs/references/networks-json.mdx:113]: Added info about providers and type

File: docs/references/networks-json.mdx (hunk 1 above)
Comment (jessiemongeon1) [reply docs/references/networks-json.mdx]: Fixed indentation

**Pull request merged**